payjoin/src/sender/mod.rs

```diff
@@ -321,28 +321,25 @@ impl Context {
                 }
                 // theirs (receiver)
                 None | Some(_) => {
-                    /* this seems to be wrong but not sure why/how
-                    match (&proposed.psbtin.final_script_sig, &proposed.psbtin.final_script_witness) {
-                        // TODO: use to compute weight correctly
-                        (Some(sig), Some(witness)) => (),
-                        _ => return Err(InternalValidationError::ReceiverTxinNotFinalized)
+                    // Verify the PSBT input is finalized
+                    ensure!(
+                        proposed.psbtin.final_script_sig.is_some()
+                            || proposed.psbtin.final_script_witness.is_some(),
+                        ReceiverTxinNotFinalized
+                    );
+                    if let Some(script_sig) = &proposed.psbtin.final_script_sig {
+                        total_weight += Weight::from_non_witness_data_size(
+                            32 + 4 + 4 + script_sig.encoded_size(),
```

Comment:
Why do you add the 40 units here, above you already added the out point size it seems

Reply:
Thanks for the review Ben 🎈

Isn't this the branch where there is indeed a script_sig? No script sig should be handled by `if let Some(script_witness)` below.

 I'm not sure which is the primary source to reference, but I'm looking at [this recent rust-bitcoin pr](https://github.com/rust-bitcoin/rust-bitcoin/pull/1467/files). I also believe it to be the outpoint (36) + sequence (4). and will document

Do you have a clue what unit tests for this code might be useful? I guess I should find some transactions with known weights and check this code against them. There are some examples in that referenced issue but I don't know where they came from.